test/scripts/products/h24-verify-selfmanaged-apicast-and-custom-policy/h24-verify-selfmanaged-apicast-and-custom-policy.go

```diff
@@ -0,0 +1,471 @@
+	if err != nil {
+		log.Error("Error create adminportal-credentials secret. Exit...", err)
+		return
+	}
+	time.Sleep(5 * time.Second)
+
+	// Use self-managed APIcast instead of the builded one for API
+	// Update apicast secrets Hosts in redhat-rhoam-3scale
+	if promoteManually != "yes" {
+		updateThreeScaleRoutesToSelfManagedApicast(ctx, client)
+	} else {
+		promoteSelfManagedAPIcastInUI()
+	}
+
+	// Install 3scale APIcast gateway operator
+	err = installThreeScaleApicastGatewayOperator(ctx, client, apicastNamespace)
+	if err != nil {
+		log.Error("Error install 3scale APIcast gateway Operator. Exit...", err)
+		return
+	}
+
+	// Create a self-managed APIcast
+	err = createSelfManagedApicast(ctx, client, apicastNamespace)
+	if err != nil {
+		log.Error("Error create self-managed APIcast. Exit...", err)
+		return
+	}
+
+	// Create a route for the self-managed APIcast
+	err = createApicastRoute(ctx, client, apicastNamespace, threeScaleAdminPortal)
+	if err != nil {
+		log.Error("Error create route for self-managed APIcast. Exit...", err)
+		return
+	}
+
+	log.Info("Self-managed APIcast API gateway - Deployed.")
+}
+
+// Create customer users in dedicated-admins group
+func createCustomerUsers(customerAdminPassword string) error {
+	log.Info("Creating customer users in dedicated-admins group")
+	command := "PASSWORD=" + customerAdminPassword + " DEDICATED_ADMIN_PASSWORD=" + customerAdminPassword + " ../../../../scripts/setup-sso-idp.sh"
+	err := run_shell_command(command)
+	return err
+}
+
+func createNamespace(ctx context.Context, client k8sclient.Client, apicastNamespace string) error {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: apicastNamespace,
+		},
+		Spec: corev1.NamespaceSpec{},
+	}
+	err := client.Get(ctx, k8sclient.ObjectKey{Name: ns.Name}, ns)
+	if err != nil {
+		log.Info("Creating namespace " + ns.Name)
+		err = client.Create(ctx, ns)
+		if err != nil {
+			log.Error("Error create namespace "+apicastNamespace, err)
+			return err
+		}
+	} else {
+		log.Info("Namespace " + apicastNamespace + " already exists")
+	}
+	err = client.Get(ctx, k8sclient.ObjectKey{Name: ns.Name}, ns)
+	if err != nil {
+		log.Error("Error get namespace"+ns.Name, err)
+		return err
+	}
+	command := "oc project " + apicastNamespace
+	return run_shell_command(command)
+}
+
+// Import APIcast Image
+func importApicastImage(tag string) error {
+	log.Info("Import APIcast Image")
+	command := "oc import-image 3scale-amp2/apicast-gateway-rhel8:" + tag + " --from=registry.redhat.io/3scale-amp2/apicast-gateway-rhel8:" + tag + " --confirm"
+	err := run_shell_command(command)
+	return err
+}
+
+// Get 3scale admin token
+func getThreeScaleAdminToken(ctx context.Context, client k8sclient.Client) (string, error) {
+	log.Info("Get 3scale admin Token")
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system-seed",
+		},
+	}
+	err := client.Get(ctx, k8sclient.ObjectKey{Name: s.Name, Namespace: "redhat-rhoam-3scale"}, s)
```

Comment:
Thanks for comment. We will use global variable instead of prefix, as "redhat-rhoam-" apears only in one case. It's differ in example for h22, where the prefix is used for more than one variable. We also avoid extra parameter. But it's a good comment, and we will improve.

Reply:
Done